features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/details/RoomMemberDetailsPresenter.kt

```diff
@@ -36,11 +44,35 @@ class RoomMemberDetailsPresenter @AssistedInject constructor(
 
     @Composable
     override fun present(): RoomMemberDetailsState {
+        val coroutineScope = rememberCoroutineScope()
+        var confirmationDialog by remember { mutableStateOf<ConfirmationDialog?>(null) }
+        var isBlocked by remember { mutableStateOf(roomMember.isIgnored) }
 
-//        fun handleEvents(event: RoomMemberDetailsEvents) {
-//            when (event) {
-//            }
-//        }
+        fun handleEvents(event: RoomMemberDetailsEvents) {
+            when (event) {
+                is RoomMemberDetailsEvents.BlockUser -> {
+                    if (event.needsConfirmation) {
+                        confirmationDialog = ConfirmationDialog.Block
```

Comment:
Please extract some methods here using extension on `CoroutineScope`

Reply:
I've tried doing that in https://github.com/vector-im/element-x-android/pull/340/commits/96bbd77fcbedcd8047c298e2a4e63c7a5ebd000b, but it's not as easy to do it as in `ChangeServerFlowPresenter`.